pom.xml

```diff
@@ -1,284 +1,329 @@
+        <version>2.5.3</version>
+        <dependencies>
+          <dependency>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-scm-plugin</artifactId>
+            <version>1.8.1</version>
+          </dependency>
+        </dependencies>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>1.4.0</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>java</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <mainClass>com.contrastsecurity.sdk.ContrastSDK</mainClass>
+          <arguments>
+          </arguments>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-gpg-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>sign-artifacts</id>
+            <phase>deploy</phase>
+            <goals>
+              <goal>sign</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-source-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>attach-sources</id>
+            <phase>package</phase>
+            <goals>
+              <goal>jar-no-fork</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <configuration>
+          <detectJavaApiLink>false</detectJavaApiLink>
+        </configuration>
+        <executions>
+          <execution>
+            <id>attach-javadocs</id>
+            <phase>package</phase>
+            <goals>
+              <goal>jar</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-deploy-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>deploy</id>
+            <phase>deploy</phase>
+            <goals>
+              <goal>deploy</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>2.19.1</version>
+        <configuration>
+          <excludes>
+            <exclude>**/ScreenerTest.java</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>8</source>
+          <target>8</target>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
```

Comment:
Is spotless better than fmt? I think [fmt](https://github.com/coveooss/fmt-maven-plugin) works on java 8.

Reply:
Both tools require Java 11:

```
[ERROR] Failed to execute goal com.coveo:fmt-maven-plugin:2.11:check (default) on project contrast-sdk-java: Execution default of goal com.coveo:fmt-maven-plugin:2.11:check failed: An API incompatibility was encountered while executing com.coveo:fmt-maven-plugin:2.11:check: java.lang.UnsupportedClassVersionError: com/google/googlejavaformat/java/FormatterException has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

A [note in the fmt readme](https://github.com/coveooss/fmt-maven-plugin#using-with-java-8) tells us that the underlying library Google Java Format has migrated to Java 11 starting in version Google Java Format version 1.8.

I am confused by this, because [that repository appears to be configured to target Java 1.8](https://github.com/google/google-java-format/blob/1c45fa55b1a483314f8cb52bd0a0b21838c2bc32/pom.xml#L97), but I can't argue with the bytecode version in the jar 🤷 

As for "is spotless better than fmt", I don't know. Spotless is more flexible, but that could be good or bad. I chose Spotless because it has plugins for both Maven and Gradle, so I like that we can be consistent across our projects instead of using Spotless for Gradle and fmt for Maven.

Reply:
ts uses 2.9 which works with 1.8

Reply:
It works with 1.8 because it uses an older version of Google Java Format. You can achieve the same effect with the latest Spotless plugin version by configuring it to use Google Java Format 1.7.

However, I don't think we want to do that, because it invites issues with developers' editors. For example, if you have the latest [IntelliJ Google Java Format plugin](https://plugins.jetbrains.com/plugin/8527-google-java-format) installed, then it will use Google Java Format 1.9 which will apply different formatting. Consequently, your IDE formatting and the build's formatting will be out of sync and they will fight each other.

Reply:
ah ok

Reply:
For what it's worth, we're running into this problem because we're using a build matrix to test the project across different versions of Java. When you do that, your entire Maven build must be compatible with each JDK version you need to test with (vs only your code).

Instead of using a build matrix, we could try to migrate the build to [verify the project using Maven Toolchains](https://johnathangilday.com/blog/maven-toolchain-tests/). My take is that this formatting issue isn't enough of a problem to migrate from the relatively simple matrix build to the relatively more complex Maven Toolchains, so we deal with it for now and we can always revisit 🙂